index.xml

```diff
@@ -8593,94 +8243,74 @@ results.numbers.length;       <span class='comment'>// So this still evaluates t
-            whose presence affects only the ECMAScript binding.
-          </p>
 
-          <div id='ArrayClass' class='section'>
-            <h4>[ArrayClass]</h4>
+          <div id='es-frozen-array' class='section'>
+            <h4>Frozen arrays — FrozenArray&lt;<var>T</var>&gt;</h4>
 
             <p>
-              If the <a class='xattr' href='#ArrayClass'>[ArrayClass]</a>
-              <a class='dfnref' href='#dfn-extended-attribute'>extended attribute</a>
-              appears on an <a class='dfnref' href='#dfn-interface'>interface</a>
-              that is not defined to <a class='dfnref' href='#dfn-inherit'>inherit</a>
-              from another, it indicates that the internal <span class='prop'>[[Prototype]]</span>
-              property of its <a class='dfnref' href='#dfn-interface-prototype-object'>interface prototype object</a>
-              will be the <span class='estype'>Array</span> prototype object rather than
-              the <span class='estype'>Object</span> prototype object.  This allows
-              <span class='estype'>Array</span> methods to be used more easily
-              with objects implementing the interface.
+              Values of frozen array types are represented by frozen ECMAScript
+              <span class='estype'>Array</span> object references.
             </p>
             <p>
-              The <a class='xattr' href='#ArrayClass'>[ArrayClass]</a> extended attribute
-              <span class='rfc2119'>MUST</span> <a class='dfnref' href='#dfn-xattr-no-arguments'>take no arguments</a>.
-              It <span class='rfc2119'>MUST NOT</span> be used on an interface that
-              has any <a class='dfnref' href='#dfn-inherited-interfaces'>inherited interfaces</a>.
+              An ECMAScript value <var>V</var> is
+              <a class='dfnref' href='#dfn-convert-ecmascript-to-idl-value'>converted</a>
+              to an IDL <span class='idltype'>FrozenArray&lt;<var>T</var>&gt;</span> value
+              by running the following algorithm:
             </p>
-            <div class='note'>
-              <p>
-                Interfaces using <a class='xattr' href='#ArrayClass'>[ArrayClass]</a> will
-                need to define a “length” <a class='dfnref' href='#dfn-attribute'>attribute</a>
-                of type <span class='estype'>unsigned long</span> that exposes the length
-                of the array-like object, in order for the inherited <span class='estype'>Array</span>
-                methods to operate correctly.  Such interfaces would typically also
-                <a class='dfnref' href='#dfn-support-indexed-properties'>support indexed properties</a>,
-                which would provide access to the array elements.
-              </p>
-            </div>
+            <ol class='algorithm'>
+              <li>Let <var>values</var> be the result of
+                <a class='dfnref' href='#dfn-convert-ecmascript-to-idl-value'>converting</a>
```

Comment:
Hmm, so in what way are they different, apart from doing the iterable check up front (before doing it again in the "convert to sequence<T>" algorithm)?

Maybe it's better to just call into the "convert to sequence<T>" and "convert to FrozenArray<T>" algorithms directly here from the union conversion algorithm. That would still have some duplicate processing (we'd check for Date/RegExp twice now, rather than checking for iterability twice) but at least would capture any changes we make to those two algorithms without needing to change anything here.